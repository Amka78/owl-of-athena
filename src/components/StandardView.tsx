import React, { FunctionComponent } from "react";
import { View, ViewStyle } from "react-native";

import { CommonStyles } from "../styles";

type StandardViewProps = {
    children: React.ReactNode;
    standardViewStyle?: ViewStyle;
};
export const StandardView: FunctionComponent<StandardViewProps> = (
    props: StandardViewProps
) => {
    return (
        <View style={CommonStyles.rootContainer}>
            <View style={[CommonStyles.standardView, props.standardViewStyle]}>
                {props.children}
            </View>
        </View>
    );
};
