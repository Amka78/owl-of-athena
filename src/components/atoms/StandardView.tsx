import React, { FunctionComponent } from "react";
import { View, ViewStyle } from "react-native";
import { useTheme } from "react-native-paper";

import { CommonStyles } from "../../styles";

type StandardViewProps = {
    children: React.ReactNode;
    standardViewStyle?: ViewStyle;
};
export const StandardView: FunctionComponent<StandardViewProps> = (
    props: StandardViewProps
) => {
    const theme = useTheme();
    return (
        <View
            style={[
                CommonStyles.rootContainer,
                { backgroundColor: theme.colors?.background },
            ]}
        >
            <View
                style={[
                    CommonStyles.standardView,
                    { backgroundColor: theme.colors?.background },
                    props.standardViewStyle,
                ]}
            >
                {props.children}
            </View>
        </View>
    );
};
