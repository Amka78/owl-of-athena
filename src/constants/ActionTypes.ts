const LOGIN = "LOGIN" as const;
const UPDATE_USER = "UPDATE_USER" as const;
const LOGOUT = "LOGOUT" as const;
const CACHE_SETTINGS = "CACHE_SETTINGS" as const;
const INIT = "INIT" as const;
const CACHE_PROFILES = "CACHE_PROFILES" as const;
const CACHE_SESSIONS = "CACHE_SESSIONS" as const;
const SELECT_SESSION = "SELECT_SESSION" as const;
const SELECT_SESSION_DETAIL = "SELECT_SESSION_DETAIL" as const;
const INITIALIZE_AURORA = "INITIALIZE_AURORA" as const;
const INITIALIZE_SESSION = "INITIALIZE_SESSION" as const;
const UPDATE_FILTER = "UPDATE_FILTER" as const;
const DELETE_SESSION = "DELETE_SESSION" as const;
const UPDATE_SESSION = "UPDATE_SESSION" as const;
export default {
    INIT,
    INITIALIZE_AURORA,
    INITIALIZE_SESSION,
    LOGIN,
    LOGOUT,
    UPDATE_USER,
    CACHE_SESSIONS,
    DELETE_SESSION,
    SELECT_SESSION,
    SELECT_SESSION_DETAIL,
    UPDATE_FILTER,
    UPDATE_SESSION,
    CACHE_SETTINGS,
    CACHE_PROFILES,
};
