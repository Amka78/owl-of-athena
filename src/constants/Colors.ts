const teal = "#00ffe6";
const purple = "#4a2fba";
const white = "#ffffff";
export default {
    black: "#000000",
    charcoal: "#333333",
    gray: "#9B9B9B",
    white: white,
    purple: purple,
    navy: "#070724",
    navy_darker: "#09092B",
    blue: "#001438",
    cyan: "#bcfeff",
    teal: teal,
    yellow: "#daf392",
    red: "#ff4310",
    light_orange: "#ff8260",
    semi_transparent: "#66000000",
    aurora_connected: teal,
    aurora_connected_text: purple,
    aurora_disconnected: purple,
    aurora_disconnected_text: white
};
