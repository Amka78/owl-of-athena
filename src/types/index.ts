//#region Import Modules
import { GuestUser } from "./Auth";
import { CreateUser, Login } from "./Login";
import { Signup } from "./Signup";
import { AuroraSound } from "./Sound";
import { User } from "./User";
import { ExperimentalNavigator, WakeLockSentinel } from "./WakeLock";
//#endregion

//#region Types
export {
    AuroraSound,
    CreateUser,
    ExperimentalNavigator,
    GuestUser,
    Login,
    Signup,
    User,
    WakeLockSentinel,
};
//#endregion
