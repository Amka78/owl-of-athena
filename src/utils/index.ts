import BaseUrl from "./BaseUrl";
export { BaseUrl };
