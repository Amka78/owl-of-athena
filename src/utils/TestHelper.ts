import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
