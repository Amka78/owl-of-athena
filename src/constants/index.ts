import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
