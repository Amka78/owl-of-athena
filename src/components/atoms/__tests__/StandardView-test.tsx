/* eslint-disable @typescript-eslint/ban-ts-comment */
//#region Import Modules
import { View } from "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";

import { createMock, toJson } from "../../../utils/TestHelper";
import { StandardView, StandardViewProps } from "../StandardView";
import { useTheme } from "react-native-paper";
import { Colors } from "../../../constants";
import { Theme } from "react-native-paper/lib/typescript/types";
//#endregion

//#region Test
let component: ShallowWrapper<StandardViewProps, unknown, unknown>;
jest.mock("react-native-paper");
const useThemeMock = useTheme as jest.Mock<Theme>;
describe("StandardView UnitTest", () => {
    afterEach(() => {
        jest.resetAllMocks();
    });

    it("Renders correctly", () => {
        // @ts-ignore
        useThemeMock.mockReturnValue({ colors: { background: Colors.navy } });
        component = createMock(
            <StandardView>
                <View></View>
            </StandardView>
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    it("Overwrite styles", () => {
        // @ts-ignore
        useThemeMock.mockReturnValue({ colors: { background: Colors.navy } });
        component = createMock(
            <StandardView
                rootViewStyle={{ marginTop: 0 }}
                standardViewStyle={{ marginTop: 0 }}
            >
                <View></View>
            </StandardView>
        );
    });
});
//#endregion
