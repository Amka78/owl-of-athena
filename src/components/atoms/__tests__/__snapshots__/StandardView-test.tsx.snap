// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardView UnitTest Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#070724",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "marginLeft": 48,
          "marginRight": 48,
        },
        Object {
          "backgroundColor": "#070724",
        },
        undefined,
      ]
    }
  >
    <View />
  </View>
</View>
`;
